### PR TITLE
Feat/add axelar executor contract

### PR DESCRIPTION
**Pull request by dobertRowneySr**
## Purpose
Add Axelar gateway executor in order to enable cross chain logic (see #8 for details).
##### Changes
- added extra function `lend` which actually performs the NFT lend
- used `callContractWithToken` so payment is taken care by Axelar (without relying on swing.xyz)
- Added some unit tests
- Testing for the cross chain behavior will be done on integration tests

I am not sure whether we want to keep the `INFTMarketplace` interface, since I guess that the `rent` and `lend` function can be made `internal`

## Checklist

- [x] Title follows the [naming convention](../README.md#pull-requests).
- [x] Add a description.
- [ ] Add screenshots (optional).
- [ ] Add testing instructions.
- [x] Add at least one reviewer.
- [ ] Apply related labels (optional).
- [ ] Leave comments on changes you would like to discuss.
- [ ] Remove changes that are not related to this pull request.
- [ ] Remove debugging code.
- [x] Test your changes locally and on staging (if applicable).
- [ ] Ensure test coverage value in config matches actual coverage (if applicable).
- [ ] Ensure the build is passing.

### CODE REVIEW

**Review by danielssonn — APPROVED** [review]
Very nice!

File: protocol/test/unit/1-test.ts [hunk 1]
```diff
@@ -157,22 +157,35 @@ describe('NFTMarket', function () {
         )
       })
 
-      it('Should increase the lenders balance', async () => {
-        const [, addr1] = await ethers.getSigners()
-        expect(await dummyCoin!.balanceOf(addr1.address)).to.equal(
-          ethers.utils.parseEther('0.001')
-        )
+      it('Should emit the NFTRented event', async () => {
+        const [, , addr2] = await ethers.getSigners()
```
Comment (mattxyzeth) [inline protocol/test/unit/1-test.ts:161]: This looks like it's not needed.

File: protocol/test/unit/1-test.ts [hunk 2]
```diff
@@ -181,6 +194,12 @@ describe('NFTMarket', function () {
 
         expect(listing.rental.renter).to.equal(addr2.address)
       })
+
+      it('Should emit the NFTLent event', async () => {
+        const [, , addr2] = await ethers.getSigners()
```
Comment (mattxyzeth) [inline protocol/test/unit/1-test.ts]: Same with this. It doesn't seem to be used in this test.

**Review by mattxyzeth — APPROVED** [review]
We'll need some more test coverage, but it looks like it will work. Have you tested on the testnets?

**Comment by dobertRowneySr** [comment]
I am closing this since this version of the protocol has problems

**Pull request closed without merging**